hs-bindgen/src-internal/HsBindgen/Frontend/Pass.hs

```diff
@@ -69,5 +71,20 @@ class IsPass (p :: Pass) where
   -- are showable, for example).
   type Ann (ix :: Symbol) p :: Star
 
+  -- | Configuration required to run the pass
+  type Config p :: Star
+
+  -- | Trace messages possibly emitted by the pass
+  type Msg p :: Star
```

Comment:
Let's make this a `data` family rather than a `type` family. Passes that provide no messages are very rare, and data families generally work better with type inference.

Reply:
Done. I created an associated data family.